knock/src/components/MainPageInput/MainPageInput.tsx

```diff
@@ -0,0 +1,61 @@
+import HumanIcon from '../../core/assets/icon/Person.svg';
+import styles from './mainPageInput.module.scss';
+import React, { useState } from 'react';
+import UButton from '../../core/ui/buttons/UButton/UButton';
+import Input from '../../core/ui/Input/Input';
+import { createSubject } from '../../lib/api/Subject';
+import { useNavigate } from 'react-router-dom';
+import useUserInfo from '../../lib/hooks/useUserInfo';
+
+interface MainPageInputProps {
+  onSubmit: (name: string) => void;
+}
+
+const MainPageInput: React.FC<MainPageInputProps> = ({ onSubmit }) => {
+  const [name, setName] = useState('');
+  const navigate = useNavigate();
+  const { handleUserInfo } = useUserInfo();
+
+  const handleNameSet = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setName(e.target.value);
+  };
+
+  const handleSubmit = async (e: React.FormEvent<HTMLElement>) => {
+    e.preventDefault();
+    try {
+      const { id, name: subName, imageSource } = await createSubject({ name });
+      navigate(`/post/${id}/answer`);
+      handleUserInfo({ id, name: subName, imageSource });
+    } catch (e) {
+      console.error(e);
+    }
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter') {
+      handleSubmit(e);
+    }
```

Comment:
빈 값일때는 handleSubmit이 작동하지 않게 하는 것도 좋아보입니다!😄
```suggestion
    if (name && e.key === 'Enter') {
      handleSubmit(e);
    }
```